opal/corelib/string/encoding.rb

```diff
@@ -277,6 +284,9 @@ def force_encoding(encoding)
 
       if (encoding === self.encoding) { return self; }
 
+      if (typeof self === 'string') {
+        self = self.$dup();
+      }
```

Comment:
we call `dup` to create a String object otherwise `Opal.set_encoding` throws an FrozenError.

Reply:
Replacing `self` is very risky and will break all other references to the string, let's keep the FrozenError and solve the issue in some other way. This would be consistent with Ruby using frozen string literals:

```rb
# frozen_string_literal: true
"asdf".force_encoding('ascii') # => FrozenError: can't modify frozen String: "asdf"
```

Reply:
If I remove this line, `USE_STRICT=true bin/rake minitest_node_nodejs` is failing:

```
TestNodejsOpalBuilder#test_should_build_simple_ruby_file:
Opal::SyntaxError: can't modify frozen String
    test/nodejs/fixtures/hello.rb:in `puts "Hello world"'
    FrozenError: can't modify frozen String
      from <internal:corelib/runtime.js>:2446:7:in `Opal.set_encoding'
      from <internal:corelib/string/encoding.rb>:287:1:in `force_encoding'
      from <internal:corelib/string.rb>:243:5:in `force_encoding'
      from parser/lexer/literal.rb:245:7:in `b'
      from parser/lexer/literal.rb:47:22:in `coerce_encoding'
      from <internal:corelib/runtime.js>:1729:5:in `Opal.send2'
      from <internal:corelib/runtime.js>:1719:5:in `Opal.send'
      from <internal:corelib/class.rb>:39:1:in `new'
      from <internal:corelib/runtime.js>:1729:5:in `Opal.send2'
      from <internal:corelib/runtime.js>:1719:5:in `Opal.send'
```

Reply:
> If the problem is calling String#b in parser we can try to fix that, especially since the current implementation is wrong and tries to force the encoding instead of copying the string:

Sounds good, I will do that 👍🏻